### PR TITLE
apm/php: document how to trace long running scripts

### PR DESCRIPTION
**Pull request by labbati**
### What does this PR do?

Since our documentation lacked a clear explanation of how to trace both short- and long-running CLI PHP applications, we decided to provide sample script and full examples for the two modes.

### Preview
[Preview](https://docs-staging.datadoghq.com/apm/php/long-running/tracing/setup_overview/setup/php/?tab=containers#tracing-cli-scripts)

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/setup_overview/setup/php.md [hunk 1]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
 When [`open_basedir`][12] setting is used, then `/opt/datadog-php` should be added to the list of allowed directories.
 When the application runs in a docker container, the path `/proc/self` should also be added to the list of allowed directories.
 
+## Tracing CLI scripts
+
+### Sort-running CLI scripts
+
+A short-running script runs typically for a few seconds or minutes and the expected behavior is to receive one trace each time the script is executed.
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:424]:
```suggestion
A short-running script typically runs for a few seconds or minutes and the expected behavior is to receive one trace each time the script is executed.
```

File: content/en/tracing/setup_overview/setup/php.md [hunk 2]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
...
+
+By default, tracing is disabled for PHP scripts that run from the command line. Opt-in by setting `DD_TRACE_CLI_ENABLED` to `true`.
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:426]:
```suggestion
By default, tracing is disabled for PHP scripts that run from the command line. Opt in by setting `DD_TRACE_CLI_ENABLED` to `true`.
```

File: content/en/tracing/setup_overview/setup/php.md [hunk 3]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
...
+
+```
+$ export DD_TRACE_CLI_ENABLED=true
+
+# Optionally, set the agent host and port if different from localhost and 8126, respectively
+$ export DD_AGENT_HOST=agent
+$ export DD_TRACE_AGENT_PORT=8126
+```
+
+Assume the following `script.php` that runs a Curl request
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:436]:
```suggestion
For example, assume the following `script.php` runs a Curl request:
```

File: content/en/tracing/setup_overview/setup/php.md [hunk 4]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
...
+sleep(1);
+
+$ch = curl_init('https://httpbin.org/delay/1');
+curl_exec($ch);
+
+sleep(1);
+
+```
+
+Run the script
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:450]:
```suggestion
Run the script:
```

File: content/en/tracing/setup_overview/setup/php.md [hunk 5]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
...
+
+```
+$ php script.php
+```
+
+One trace is generated and sent to the Datadog backend when the script terminates.
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:456]:
```suggestion
Once run, the trace is generated and sent to the Datadog backend when the script terminates.
```

File: content/en/tracing/setup_overview/setup/php.md [hunk 6]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
...
+
+{{< img src="tracing/setup/php/short-running-cli.jpg" alt="Trace for a short running PHP CLI script" >}}
+
+### Long-running CLI scripts
+
+A long-running script runs for hours or days. Typically, such scripts execute repetitively a specific task, for example processing new incoming messages or new lines added to a table in a database. The expected behavior is that one trace is generated for each "unit of work", for example the processing of a message.
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:462]:
```suggestion
A long-running script runs for hours or days. Typically, such scripts repetitively execute a specific task, for example processing new incoming messages or new lines added to a table in a database. The expected behavior is that one trace is generated for each "unit of work", for example the processing of a message.
```

File: content/en/tracing/setup_overview/setup/php.md [hunk 7]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
...
+
+By default, tracing is disabled for PHP scripts that run from the command line. Opt-in by setting `DD_TRACE_CLI_ENABLED` to `true`.
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:464]:
```suggestion
By default, tracing is disabled for PHP scripts that run from the command line. Opt in by setting `DD_TRACE_CLI_ENABLED` to `true`.
```

File: content/en/tracing/setup_overview/setup/php.md [hunk 8]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
...
+
+# Optionally, set service name, env, etc...
+$ export DD_SERVICE=my_service
+
+# Optionally, set the agent host and port if different from localhost and 8126, respectively
+$ export DD_AGENT_HOST=agent
+$ export DD_TRACE_AGENT_PORT=8126
+```
+
+Assume the following `long_running.php`
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:480]:
```suggestion
For example, assume the following `long_running.php` script:
```

File: content/en/tracing/setup_overview/setup/php.md [hunk 9]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
...
+/** A loop that runs forever waiting for new messages */
+while (true) {
+    $messages = waitForNewMessages();
+    foreach ($messages as $message) {
+        processMessage($message, $processors);
+    }
+}
+```
+
+Run the script
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:572]:
```suggestion
Run the script:
```

File: content/en/tracing/setup_overview/setup/php.md [hunk 10]
```diff
@@ -417,6 +417,168 @@ Note that `DD_TRACE_RESOURCE_URI_MAPPING_INCOMING` applies to only incoming requ
...
+
+```
+$ php long_running.php
+```
+
+One trace is generated and sent to the Datadog backend every time a new message is processed.
```
Comment (sarina-dd) [inline content/en/tracing/setup_overview/setup/php.md:578]:
```suggestion
Once run, one trace is generated and sent to the Datadog backend every time a new message is processed.
```

**Comment by labbati** [comment]
Thank you for all the improvements @sarina-dd

**Review by sarina-dd — APPROVED** [review]

**Pull request merged**